index.html

```diff
@@ -2038,32 +2052,37 @@ <h3>Verify Proof</h3>
         <h3>Verify Proof Sets and Chains</h3>
         <p>
 In a proof set or proof chain, a <a>secured data document</a> has a
-<code>proof</code> attribute which contains a list of proofs
+<code>proof</code> attribute which contains a list of proofs 
 (<var>proof_list</var>).
 The following algorithm specifies how to check the authenticity and
 integrity of a <a>secured data document</a> by verifying each
 proof in the <var>proof_list</var>. Required inputs are a
-<a>secured data document</a> (<var>securedDocument</var>). A list of
+<a>secured data document</a> (<var>securedDocument</var>). A list of 
 verification results is produced as output.
         </p>
         <ol class="algorithm">
           <li>
-Set <var>proof_list</var> to </var><var>securedDocument</var>.<var>proof</var>.
+Set <var>proof_list</var> to </var><var>securedDocument</var>.<var>proof</var>. 
           </li>
           <li>
-For each <var>proof</var> in <var>proof_list</var>, run steps 2&ndash;11 of the algorithm in
-section <a href="verify-proof"></a>, and if no exceptions are raised, associate
+For each <var>proof</var> in <var>proof_list</var>, run steps 2 through 11 of the algorithm in
+section <a href="verify-proof"></a>; if no exceptions are raised, associate
 the <var>isProofVerified</var> value with this <var>proof</var>.
           </li>
           <li>
-For each <var>proof</var> in <var>proof_list</var> that contains a
-<code>previousProof</code> attribute, modify the associated
-<var>isProofVerified</var> by iteratively checking whether the proof identified by
-the <code>id</code> value in <var>previousProof</var> is valid, along with any
-<code>previousProofs</code> up the chain.
+For each <var>proof</var> in <var>proof_list</var> that contains a 
+<code>previousProof</code> attribute, modify the associated 
+<var>isProofVerified</var> by iteratively checking that the proof identified by
+the <code>id</code> value in <var>previousProof</var>, along with any 
+<code>previousProofs</code> up the chain, is valid. If <code>previousProof</code> is an
+array, then all previous proof proof chains identified by the array elements 
+MUST be valid for the current proof to be considered valid. Cycle detection, 
+such as keeping a list of previously visited proofs, MUST be used to guard 
+against malicious proofs. If a cycle is detected in a proof chain, 
+then a `PROOF_CHAIN_CYCLE_ERROR` MUST be raised.
           </li>
           <li>
-Return the <var>proof_list</var> along with each proof's associated
+Return the <var>proof_list</var> along with each proofs associated 
```

Comment:
```suggestion
Return the <var>proof_list</var> along with each proof's associated 
```
The original phrasing with the apostrophe, is correct.